lib/chef/provider/apt_update.rb

```diff
@@ -62,16 +58,17 @@ def apt_up_to_date?
 
       def do_update
         [STAMP_DIR, APT_CONF_DIR].each do |d|
-          build_resource(:directory, d, caller[0]) do
+          declare_resource(:directory, d, caller[0]) do
             recursive true
-          end.run_action(:create)
+          end
         end
 
-        build_resource(:file, "#{APT_CONF_DIR}/15update-stamp", caller[0]) do
+        declare_resource(:file, "#{APT_CONF_DIR}/15update-stamp", caller[0]) do
           content "APT::Update::Post-Invoke-Success {\"touch #{STAMP_DIR}/update-success-stamp 2>/dev/null || true\";};"
-        end.run_action(:create_if_missing)
+          action :create_if_missing
+        end
 
-        shell_out!("apt-get -q update")
+        declare_resource(:execute, "apt-get -q update", caller[0])
```

Comment:
We don't have the Resources DSL mixed into core chef providers.  We tore it out of Chef 12.  There was one reason which was because the errors back from method missing were horrid.  We no longer use method_missing (or won't in Chef-13) so that rationale will be going away.  I thought that there was also some other use case that @danielsdeleo determined where having the DSL mixed into core providers allowed cookbooks to troll the internals of core providers.  I have forgotten the specifics of that, but until we dig that up again I'm not very inclined to mixin the DSL into core providers -- particularly in light of all the confusion caused by the 12.5 resource syntax.  For now I'm liking the separation that declare_resource provides.